cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/StateVariablesExport.java

```diff
@@ -125,19 +131,161 @@ private static void buildAngleRefs(Bus bus, Map<String, String> angleRefs, Cgmes
+            }
+            boolean isInAccordance;
+            if (hasAnyFinite(busViewBus, Terminal::getP) && hasAnyFinite(busViewBus, Terminal::getQ)) {
+                double sumP = sumTerminals(busViewBus, Terminal::getP);
+                double sumQ = sumTerminals(busViewBus, Terminal::getQ);
+                isInAccordance = Math.abs(sumP) < maxPMismatchConverged && Math.abs(sumQ) < maxQMismatchConverged;
+                if (!isInAccordance && LOG.isInfoEnabled()) {
+                    LOG.info("Bus {} is not in accordance with Kirchhoff's first law. Mismatch = {}", bus, String.format("(%.4f, %.4f)", sumP, sumQ));
+                    logDetail(busViewBus);
+                    LOG.debug(String.format("  %7.2f  %7.2f  Sum", sumP, sumQ));
+                }
+            } else {
+                isInAccordance = false;
+                LOG.info("Bus {} is not in accordance with Kirchhoff's first law. All connected terminals have invalid values", bus);
+                logDetail(busViewBus);
+            }
+            checkedBusViewBuses.add(busViewBus);
+            return isInAccordance;
+        }
+
+        private boolean isSlack(Bus bus) {
+            SlackTerminal st = bus.getVoltageLevel().getExtension(SlackTerminal.class);
+            return st != null && !st.isEmpty() && st.getTerminal().getBusView().getBus() == bus;
+        }
+
+        private boolean isAngleReference(Bus bus) {
+            ReferenceTerminals reft = bus.getNetwork().getExtension(ReferenceTerminals.class);
+            if (reft == null) {
+                return false;
+            } else {
+                Set<Terminal> refts = reft.getReferenceTerminals();
+                return bus.getConnectedTerminalStream().anyMatch(refts::contains);
+            }
+        }
+
+        private boolean hasAnyFinite(Bus bus, Function<Terminal, Double> value) {
+            return bus.getConnectedTerminalStream().map(value).anyMatch(Double::isFinite);
+        }
+
+        private double sumTerminals(Bus bus, Function<Terminal, Double> value) {
+            return bus.getConnectedTerminalStream()
+                    .map(value)
+                    .filter(pq -> !Double.isNaN(pq))
+                    .mapToDouble(Double::valueOf)
+                    .sum();
+        }
+
+        private void logDetail(Bus bus) {
+            if (LOG.isDebugEnabled()) {
+                bus.getConnectedTerminalStream()
+                        .forEach(t -> LOG.info(String.format("  %7.2f  %7.2f  %s %s %s", t.getP(), t.getQ(), t.getConnectable().getType(), t.getConnectable().getNameOrId(), t.getConnectable().getId())));
+            }
+        }
+
+        Optional<Bus> getBusViewBus(Bus bus) {
+            if (bus.getVoltageLevel().getTopologyKind().equals(TopologyKind.BUS_BREAKER)) {
+                return Optional.of(bus.getVoltageLevel().getBusView().getMergedBus(bus.getId()));
+            } else {
+                if (bus.getConnectedTerminalCount() > 0) {
```

Comment:
Mmmm a bus without any connected terminal is not a bus in iidm topology computation, no?

Reply:
At node/breaker level, the computed bus/breaker buses may not have terminals because the lie in nodes connected through retained switches. The following diagram shows an example (taken from a real-world network, the model is defined as node/breaker):

<img width="1075" alt="Screenshot 2023-11-28 at 16 13 27" src="https://github.com/powsybl/powsybl-core/assets/3374819/371436ef-7637-4285-958b-983cdc8dbaa8">

The nodes 0, 15, 20 are grouped in a calculated bus/breaker bus that do not have terminals (only ends of retained switches); the same for 4, 16, 24. The color of the nodes is the indication of the scope of the bus/breaker bus. To locate the corresponding `BusView` bus we can not go through terminals, we find it checking which bus view bus from the voltage level contains the bus/breaker view bus.